File: Source/JavaScript/model/interactions/IntervalInteractionTrigger.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { InteractionTrigger } from './InteractionTrigger';

/**
 * An interaction started by an elapsed interval, normalized to seconds so nothing downstream carries a unit.
 */
export interface IntervalInteractionTrigger extends InteractionTrigger {
    seconds: number;
}

export const IntervalInteractionTriggerPropertyNames: (keyof IntervalInteractionTrigger)[] = ['seconds'];

File: Source/DotNET/Model/Interactions/IntervalInteractionTrigger.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Interactions;

/// <summary>
/// Represents an interaction started by an elapsed interval.
/// </summary>
/// <param name="Seconds">How long elapses between occurrences, normalized to seconds.</param>
/// <remarks>
/// Normalized at conversion so nothing downstream has to carry a unit or convert one. A renderer schedules
/// seconds; it does not need to know the document said 'interval 5 minutes'.
/// </remarks>
public record IntervalInteractionTrigger(int Seconds) : InteractionTrigger;
